projects/notebooks/visualizations/rhods-notebooks-performance/analyze/__init__.py

```diff
@@ -1,17 +1,60 @@
 import json
 import logging
+import numpy as np
 
-import matrix_benchmarking.common as common
+from typing import List
 
+import matrix_benchmarking.common as common
+import matrix_benchmarking.regression as regression
+import matrix_benchmarking.regression.zscore as zscore
 
 def run():
-    logging.info(f"Received {common.Matrix.count_records()} new entries")
-    for entry in common.Matrix.all_records():
-        pass
 
     logging.info(f"Received {common.LTS_Matrix.count_records()} historic LTS entries")
-    for lts_entry in common.LTS_Matrix.all_records(): pass
+    lts_entries = []
+    # Take the first result if they were gathered
+    for entry in common.LTS_Matrix.all_records():
+        if entry.results and type(entry.results) is list:
+            entry.results = entry.results[0]
+            lts_entries.append(entry)
+        else:
+            lts_entries.append(entry)
+
+    logging.info(f"Received {common.Matrix.count_records()} new entries")
 
     number_of_failures = 0
+    settings_to_check = ["version", "repeat"]
+    for entry in common.Matrix.all_records():
+        regression_results_dest = entry.location / "regression.json"
+        regression_results: List[models.RegressionResult] = []
+        for check_setting in settings_to_check:
+            controlled_settings = entry.get_settings()
+
+            try:
+                controlled_settings.pop(check_setting)
+            except KeyError:
+                logging.warning(f"Couldn't find {check_setting} setting for entry={entry.location}, skipping...")
+                continue
+
+            controlled_lts_entries = list(
+                filter(
+                    lambda x: regression.dict_part_eq(controlled_settings, x.get_settings()),
+                    lts_entries
+                )
+            )
```

Comment:
this part would be better inside the regression object. You pass all the LTS entries and the settings to check, and inside the class you do the filtering

Reply:
Either way works for me! This functionality is already inside the regression object, I essentially just copied it here per -> https://github.com/openshift-psap/topsail/pull/186#discussion_r1466479723

Reply:
oh, sorry, I'm going back and forth, sorry sorry sorry 🤦🏻